test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -70,6 +70,78 @@ public void It_builds_the_library_twice_in_a_row()
         }
 
         [Fact]
+        public void All_props_and_targets_add_themselves_to_MSBuildAllTargets()
+        {
+            List<string> expectedAllProjects = new List<string>();
+            string baseIntermediateDirectory = null;
+
+            var allProjectsFromProperty = GetValuesFromTestLibrary("MSBuildAllProjects", getValuesCommand =>
+            {
+                baseIntermediateDirectory = getValuesCommand.GetBaseIntermediateDirectory().FullName;
+
+                string preprocessedFile = Path.Combine(getValuesCommand.GetOutputDirectory("netstandard1.5").FullName, "preprocessed.xml");
+
+                //  Preprocess the file, and then scan it to find all the project files that were imported.  The preprocessed output
+                //  includes comments with long lines of "======", between which is the import statement and then the path to the file
+                //  that was imported
+                getValuesCommand.Execute("/pp:" + preprocessedFile)
+                    .Should()
+                    .Pass();
+
+                string previousLine = null;
+                bool insideDelimiters = false;
+                int lineNumber = 0;
+                foreach (string line in File.ReadAllLines(preprocessedFile))
+                {
+                    lineNumber++;
+                    if (line.All(c => c == '=') && line.Length >= 80)
+                    {
+                        if (insideDelimiters)
+                        {
+                            if (!previousLine.Trim().Equals("</Import>", StringComparison.OrdinalIgnoreCase))
+                            {
+                                expectedAllProjects.Add(previousLine);
+                            }
+                        }
+                        insideDelimiters = !insideDelimiters;
+                    }
+
+                    previousLine = line;
+                }
+
+                File.Delete(preprocessedFile);
+            }, valueType: GetValuesCommand.ValueType.Property);
+
+            string dotnetRoot = Path.GetDirectoryName(RepoInfo.DotNetHostPath);
+
+            expectedAllProjects = expectedAllProjects.Distinct().ToList();
+
+            var expectedBuiltinProjects = expectedAllProjects.Where(project => project.StartsWith(dotnetRoot, StringComparison.OrdinalIgnoreCase)).ToList();
+            var expectedIntermediateProjects = expectedAllProjects.Where(project => project.StartsWith(baseIntermediateDirectory, StringComparison.OrdinalIgnoreCase)).ToList();
+            var expectedOtherProjects = expectedAllProjects
+                .Except(expectedBuiltinProjects)
+                .Except(expectedIntermediateProjects)
+                //  TODO: Remove this when https://github.com/NuGet/Home/issues/3851 is fixed
```

Comment:
> File bug on SDK to represent the TODO and link to the blocking external issue.

To clarify, you mean file a bug in the SDK repo to remove the workaround in tests once the NuGet bug is fixed.  Correct?